analyses/1.Transfer_Sepsis_to_TB.ipynb

```diff
@@ -0,0 +1,425 @@
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "808\n",
+      "520\n",
+      "907\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(erm_tb_preds)\n",
+    "print(irm_tb_preds)\n",
+    "print(len(tb_dataset))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "ERM trained network tb accuracy: 0.7375964718853363\n",
+      "ERM trained network tb loss: 0.06897230209944204\n",
+      "IRM trained network tb accuracy: 0.6008820286659317\n",
+      "IRM trained network tb loss: 0.04761558565200637\n"
+     ]
+    }
+   ],
+   "source": [
+    "print('ERM trained network tb accuracy: {}'.format(erm_acc))\n",
+    "print('ERM trained network tb loss: {}'.format(avg_erm_loss))\n",
+    "print('IRM trained network tb accuracy: {}'.format(irm_acc))\n",
+    "print('IRM trained network tb loss: {}'.format(avg_irm_loss))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Find percent healthy\n",
+    "Our base rate is the percentage of healthy samples in the test data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Prercent healthy = 0.23042998897464168\n"
+     ]
+    }
+   ],
+   "source": [
+    "total_healthy = 0\n",
+    "for test_batch in tb_loader:\n",
+    "    expression, labels, ids = test_batch\n",
+    "    for label in labels:\n",
+    "        if label == label_to_encoding['healthy']:\n",
+    "            total_healthy += 1\n",
+    "\n",
+    "print('Prercent healthy = {}'.format(total_healthy / len(tb_dataset)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Save Results"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with open('../logs/sepsis_erm_results.pkl', 'wb') as out_file:\n",
+    "    pickle.dump(erm_results, out_file)\n",
+    "with open('../logs/sepsis_irm_results.pkl', 'wb') as out_file:\n",
+    "    pickle.dump(irm_results, out_file)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Conclusion\n",
+    "\n",
+    "While they don't perform as well on the transferred domain as the source domain, both the erm and irm trained networks have better than random (and better than the frequency of the more common class). Interestingly, the ERM trained model predicted tuberculosis 8/9 of the time, while the IRM trained model predicted tb about 5/9 of the time. The true frequency of tb in the dataset is 77%.\n",
+    "\n",
+    "These results are evidence that domain adaptation works to some extent on gene expression data even between diseases."
```

Comment:
Neat! I agree with your conclusions.

It's been a while since I looked at your code so I don't remember - are you doing anything (e.g. reweighting your loss function) to correct for class imbalance? You could try using the `pos_weight` argument to `BCEWithLogitsLoss` to do this (example from my code [here](https://github.com/greenelab/netreg/blob/fc3443a59601071dceb8e174130de524cf4dd0d3/utilities/classify_pytorch.py#L171)).

Reply:
Yep, since there's a big class imbalance I'm adjusting pos_weight to account for it. The way I'm doing this is slightly different in IRM and ERM though, which may contribute to the different predicted class ratios when doing domain adaptation.